src/sagas/wallet.js

```diff
@@ -0,0 +1,550 @@
+    config.setWalletServiceBaseUrl(WALLET_SERVICE_MAINNET_BASE_URL);
+    config.setWalletServiceBaseWsUrl(WALLET_SERVICE_MAINNET_BASE_WS_URL);
+
+    wallet = new HathorWalletServiceWallet({
+      requestPassword: showPinScreenForResult,
+      seed: words,
+      network
+    });
+  } else {
+    const connection = new Connection({
+      network: networkName, // app currently connects only to mainnet
+      servers: ['https://mobile.wallet.hathor.network/v1a/'],
+    });
+
+    const walletConfig = {
+      seed: words,
+      store: STORE,
+      connection,
+      beforeReloadCallback: () => dispatch(onStartWalletLock()),
+    };
+    wallet = new HathorWallet(walletConfig);
+  }
+
+  yield put(setWallet(wallet));
+
+  // Setup listeners before starting the wallet so we don't lose messages
+  const walletListenerThread = yield fork(setupWalletListeners, wallet);
+
+  // Create a channel to listen for the ready state and
+  // wait until the wallet is ready
+  const walletReadyThread = yield fork(listenForWalletReady, wallet);
+
+  // Store the unique device id on redux
+  yield put(setUniqueDeviceId(uniqueDeviceId));
+
+  try {
+    yield call(wallet.start.bind(wallet), {
+      pinCode: pin,
+      password: pin,
+    });
+  } catch (e) {
+    if (useWalletService) {
+      // Wallet Service start wallet will fail if the status returned from
+      // the service is 'error' or if the start wallet request failed.
+      // We should fallback to the old facade by storing the flag to ignore
+      // the feature flag
+      yield call(featureFlags.ignoreWalletServiceFlag.bind(featureFlags));
+
+      // Restart the whole bundle to make sure we clear all events
+      NativeModules.HTRReloadBundleModule.restart();
+    }
+  }
+
+  walletUtil.storePasswordHash(pin);
+  walletUtil.storeEncryptedWords(words, pin);
+  setKeychainPin(pin);
+
+  yield put(setServerInfo({
+    version: null,
+    network: networkName,
+  }));
+
+  // Wallet might be already ready at this point
+  if (!wallet.isReady()) {
+    const { error } = yield race({
+      success: take(types.WALLET_STATE_READY),
+      error: take(types.WALLET_STATE_ERROR),
+    });
+
+    if (error) {
+      yield put(startWalletFailed());
+      return;
+    }
+  }
+
+  try {
+    yield call(loadTokens);
+  } catch (e) {
+    console.log('Captured error: ', e);
+    yield put(startWalletFailed());
+    return;
+  }
+
+  const featureFlagsThread = yield fork(listenForFeatureFlags, featureFlags);
+
+  yield put(startWalletSuccess());
+
+  // The way the redux-saga fork model works is that if a saga has `forked`
+  // another saga (using the `fork` effect), it will remain active until all
+  // the forks are terminated. You can read more details at
+  // https://redux-saga.js.org/docs/advanced/ForkModel
+  // So, if a new START_WALLET_REQUESTED action is dispatched, we need to cleanup
+  // all attached forks (that will cause the event listeners to be cleaned).
+  while (true) {
```

Comment:
Why do you need this `while true` here? We execute this only once, right?

Reply:
We don't! Thanks

Fixed in aa0a15f8bebfe4cd676c5e18305e4a3784592cb6